ext/puma_http11/puma_http11.c

```diff
@@ -64,6 +66,75 @@ struct common_field {
 	VALUE value;
 };
 
+enum enum4 {
+    status1,
+    status2,
+    status3,
+    status4
+};
+
+#if defined(_WIN32)
+#include <windows.h>
+typedef HANDLE rb_nativethread_id_t;
+#else
+#include <pthread.h>
+typedef pthread_t rb_nativethread_id_t;
+#endif
+
+// Temporary workaround for https://bugs.ruby-lang.org/issues/13632
+#ifdef BROKEN_RUBY
+typedef struct half_thread {
+    void *vmlt_node1;
+    void *vmlt_node2;
+    void *self;
+    void *vm;
+    void *stack;
+    size_t stack_size;
+    void *cfp;
+    int safe_level;
+    int raised_flag;
+    void *last_status;
+    int state;
+    int waiting_fd;
+    void *passed_block;
+    void *passed_bmethod_me;
+    void *passed_ci_or_calling;
+    void *top_self;
+    void *top_wrapper;
+    #ifndef VERSION_2_4_1
+        void *base_block;
+    #endif
+    void *root_lep;
+    void *root_svar;
+    rb_nativethread_id_t thread_id;
+#ifdef NON_SCALAR_THREAD_ID
+    char thread_id_string[sizeof(rb_nativethread_id_t) * 2 + 3];
+#endif
+    enum enum4 status;
+    int to_kill;
+    int priority;
+#if defined(_WIN32)
+    void *interrupt_event;
+#elif defined(HAVE_PTHREAD_H)
+    void *ubf_list1_or_signal_thread_list;
+#ifndef VERSION_2_2_7
```

Comment:
Is the reason we're not patching other Ruby versions the little weird subtleties like this one? Basically our struct here has to exactly match the thread struct of the Ruby version we're patching, right?